tests/functional/deps/test_deps_with_vars.py

```diff
@@ -0,0 +1,67 @@
+"""Test that dbt deps works when vars are used in dbt_project.yml without defaults."""
```

Comment:
It'd probably be good to add a test here, or in another file that feels appropriate, that when doing a `dbt run` on this same setup the appropriate error gets raised 🙂